lib/matplotlib/axes/_base.py

```diff
@@ -1635,16 +1635,28 @@ def relim(self, visible_only=False):
         # Collections are deliberately not supported (yet); see
         # the TODO note in artists.py.
         self.dataLim.ignore(True)
-        self.dataLim.set_points(mtransforms.Bbox.null().get_points())
-        self.ignore_existing_data_limits = True
+        # check if there is a need to create a null bbox
+        if (self.conditionCheck()):
+            self.dataLim.set_points(mtransforms.Bbox.null().get_points())
+            self.ignore_existing_data_limits = True
 
-        for line in self.lines:
-            if not visible_only or line.get_visible():
-                self._update_line_limits(line)
+            for line in self.lines:
+                if not visible_only or line.get_visible():
+                    self._update_line_limits(line)
+ 
+            for p in self.patches:
+                if not visible_only or p.get_visible():
+                    self._update_patch_limits(p)
 
-        for p in self.patches:
-            if not visible_only or p.get_visible():
-                self._update_patch_limits(p)
+    def conditionCheck(self):
+        """
+        check if there is no patches and lines need 
+        to be changed, remain the original dataLim
+        """
+        if ((self.lines == []) and (self.patches == [])):
```

Comment:
I prefer to see this duck typed. `if not self.lines and not self.patches:` makes no requirement that each be a list (can be a tuple, set or any other iterable).